universal-login-relayer/src/core/services/EmailService.ts

```diff
@@ -0,0 +1,87 @@
+import nodemailer, {SentMessageInfo} from 'nodemailer';
+import {MailOptions} from 'nodemailer/lib/json-transport';
+import Mail from 'nodemailer/lib/mailer';
+import {CannotSendEmail} from '../utils/errors';
+
+export class EmailService {
+  private transporter: Mail;
+  constructor(transporterEmail: string, transporterEmailPassword: string) {
+    this.transporter = nodemailer.createTransport({
+      service: 'gmail',
+      auth: {
+        user: transporterEmail,
+        pass: transporterEmailPassword,
+      },
+    });
+  }
+
+  async sendMail(mailOptions: MailOptions): Promise<SentMessageInfo> {
+    try {
+      return this.transporter.sendMail(mailOptions);
+    } catch (error) {
+      throw new CannotSendEmail(error.message);
+    }
+  }
+
+  async sendConfirmationMail(email: string, code: string): Promise<SentMessageInfo> {
+    const mailOptions = {
+      from: 'UniLogin <noreply.confirmation@unilogin.io>',
+      to: email,
+      subject: 'Email Confirmation',
+      text: `To make sure your UniLogin account is safe and secure,
+      we ask you to authenticate your email address by copying the code below and pasting it in UniLogin.
+      Here is your confrimation code: ${code}`,
+      html: `
```

Comment:
We should extract HTML somewhere

Reply:
Btw what is the difference between text and HTML? Which one can we see in the mail?

Reply:
In theory both of them, some emails (like wp) enables option to view text, but by default they show html

Reply:
Done